http_output.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import( "fmt"
+	"bytes"
+	"log"
+	"net/http"
+	"time"
+	"os"
+	"os/signal"
+	"syscall"
+	"crypto/tls"
+	"strings"
+)
+
+type HttpFormatter interface {
+	Format( string ) string
+}
+
+/* This is an implementation of the HttpFormatter interface which is specific to the SensorAlertsApi */ 
+type SensorAlertsFormatter struct {
+}
+
+/* this function takes the json string specified by the event parameter and transforms it into a format
+   which the SensorAlertsApi can consume */
+func ( this *SensorAlertsFormatter ) Format( event string ) string {
+	return `{ "service" : "carbonblack", "alerts" : [ ` + event + ` ] }`
+}
+
+
+
+/* This is the HTTP implementation of the OutputHandler interface defined in main.go */
+type HttpOutput struct {
+	dest string
+	eventSentAt time.Time
+	headers map[string]string
+	ssl bool
+
+	client *http.Client
+	request *http.Request
+	formatter HttpFormatter
+}
+
+type HttpStatistics struct {
+	LastSendTime time.Time `json:"last_send_time"`
+	Destination  string    `json:"destination"`
+}
+
+/* Construct the HttpOutput object */
+func ( this *HttpOutput ) Initialize( params string ) error {
+
+        this.headers = make( map[string]string )
+
+	/* parse the parameters */
+        args := strings.Split( params, ";" )
+        this.dest = args[0]
+
+	/* add authorization token, if applicable */
+        if len( args ) > 1 {
+                this.headers["Authorization"] = "Bearer " + args[1]
```

Comment:
The authentication tokens should be refactored into their own section in the configuration file (this looks like OAuth 2.0 bearer tokens?)